libc/arc4random.c

```diff
@@ -13,8 +14,6 @@
 unsigned int arc4random(void)
 {
     unsigned int r;
-
-    while (!_rdrand32_step(&r))
```

Comment:
Don't change the logic; this keeps retrying the intrinsic call to `rdrand` until it has sufficient entropy to return a value.

Reply:
is this means we need to check the randomness of the number generated. i hope the function rdrand return a nonzero random value always. the **sufficient entropy**  means check the r value is nonzero and a complete different value always?

Reply:
The intrinsic returns the state of the carry flag which is used to detect if sufficient entropy was available. r is populated via the pointer argument.

Reply:
@ihaller is correct, the implementation for `rdrand` should check the carry flag (CF) which is 1 if sufficient entropy was found and 0 otherwise. 

Based on the [Intel SDM](https://software.intel.com/sites/default/files/managed/7c/f1/253667-sdm-vol-2b.pdf), `rdrand` will also zero the output if there is insufficient entropy, so the check you propose will look like it works, but it is subtly incorrect because it is _possible_ that the random value is actually just 0 when sufficient entropy exists.

Reply:
Made implementation to return value only if CF is set ,as shown below.
uint64_t _rdrand(void)
{
    uint64_t r;
    __asm__ volatile(
        "1%=: \n\t"
        "rdrand %0\n\t"
        "jnc 1%=\n\t"
        : "=r"(r)
        :
        :);

    return r;
}
Is it fine?

Reply:
On testing _rdrand found that it will also return 0 as random number,so  if consecutively two zeroes
come we planned to fail the test.
 
```
 for (uint64_t i = 0; i < MAX_LOOP_SIZE; i++) 
    {
        rand_num = _rdrand();                                          
        if (rand_num == 0)
        {                                                              
            rand_num = _rdrand();                                      
            OE_TEST(rand_num != 0);                                    
        }
    }
```
is this implementation okay ?